exercises/grep/canonical-data.json

```diff
@@ -332,6 +363,54 @@
             "files": ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
           },
           "expected": []
+        },
+        {
+          "description": "Multiple files, several matches, line number and file flags",
+          "property": "grep",
+          "input": {
+            "pattern": "who",
+            "flags": ["-n", "-l"],
+            "files": ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
+          },
+          "expected": [
+            "2:His wrath pernicious, who ten thousand woes",
```

Comment:
if wanting to match the behaviour of `grep`, it seems this needs to be `["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]`, as seen in the following experiment

```
$ grep -n -l hello <(echo "hello\nworld") <(echo "a\nhello") <(echo "hello\nhello")
/proc/self/fd/11
/proc/self/fd/12
/proc/self/fd/13
```

Reply:
Yes, as mentioned before I followed the behavior described in the issue which was incorrect. However this should be updated to `["iliad.txt", "paradise-lost.txt"]` since midsummer-night.txt doesn't contain the string "who". I'll update it as well.